dashboard/api_test.py

```diff
@@ -754,16 +751,17 @@ def setUp(self):
         self.expected_programs = [self.program_non_fin_aid, self.program_fin_aid]
         self.edx_client = MagicMock()
 
-    @patch('dashboard.api.get_student_enrollments', autospec=True)
-    @patch('dashboard.api.get_student_certificates', autospec=True)
-    @patch('dashboard.api.get_student_current_grades', autospec=True)
-    def test_format(self, mock_grades, mock_cert, mock_enr):
+    @patch('dashboard.api_edx_cache.CachedEdxUserData.update_cache_if_expired', new_callable=MagicMock)
+    @patch('dashboard.api_edx_cache.CachedEdxUserData.get_cached_edx_data', new_callable=MagicMock)
+    def test_format(self, mock_cache_get, mock_cache_refresh):
         """Test that get_user_program_info fetches edx data and returns a list of Program data"""
         result = api.get_user_program_info(self.user, self.edx_client)
 
-        for mocked_edx_get_function in [mock_grades, mock_cert, mock_enr]:
-            assert mocked_edx_get_function.call_count == 1
-            assert mocked_edx_get_function.call_args[0] == (self.user, self.edx_client)
+        assert mock_cache_refresh.call_count == 3
```

Comment:
Can you use `len(CachedEdxUserData.SUPPORTED_CACHES)` in place of `3` here, just to make the connection more explicit?